doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -2,7 +2,615 @@
+configuration scheme.
+Refer to the [Troubleshooting](#troubleshooting) section if something goes wrong.
+
+## Retrieving collected data
+
+Now it is time to stored temperature from the server.
+
+1. Grab application token. It is a token that you can copy from the main window
+    of the application the Administration UI.
+
+    ![Generate SDK](attach/app_token.png)
+1. Login to your machine running Kaa server (in case you are using sandbox
+    default username and password are *kaa / kaa*).
+1. Start MongoDB shell
+
+        mongo kaa
+1. Retrieve data using application token.
+
+        db.logs_$your_application_token$.find()
+
+## Changing sample period
+
+It is time to tune sampling period on a server and see what happens on the
+endpoint.
+
+1. Make sure your client application is running and sampling temperature
+1. Login to the **Admin UI** as a Tenant Developer and proceed to the **Endpoint Groups**
+    section of your application.
+
+    ![Endpoint groups](attach/endpoint_group.png)
+1. Click on the endpoint group **All** and select configuration schema from
+    **Configurations** section.
+
+    ![Endpoint groups inside](attach/endpoint_group_inside.png)
+1. Click on the latest **Configuration schema** and activate **Draft** tab. Change
+    sample period to, say, 5 seconds and hit **Save** button.
+
+    ![Endpoint groups inside](attach/new_draft_sample_period.png)
+1. Now, activate a draft by clicking **Activate** button and see client output.
+
+        Sampled temperature: 32
+        Sampled temperature: 26
+        Sampled temperature: 29
+        Sampled temperature: 33
+        Sampled temperature: 26
+        Received configuration data. New sample period: 5
+        Sampled temperature: 33
+        Sampled temperature: 30
+        Sampled temperature: 34
+        Sampled temperature: 25
+
+    You must notice that sampling period is  equal to the new
+    sample period we set in **Admin UI**
+
+
+# Next steps
+
+To create a real-world IoT solution, you will most likely need to implement more
+features into your application.
+
+Kaa provides you with practically everything you might need.
+The following overview will help you grasp the scope of Kaa capabilities
+as well as get familiar with the essential documentation,
+such as Programming guide and Administration UI guide.
+
+## Profiling and grouping
+
+During a new endpoint registration, Kaa creates an associated _endpoint profile_
+for the endpoint. An endpoint profile is basically some meaningful information
+about the endpoint which may be useful for specific applications.
+Profiles may contain things like an OS version, amount of RAM, average battery life,
+type of network connection, device operation mode – virtually anything.
+
+An endpoint profile structure in Kaa is configured using a client-side endpoint
+[LINK DOESN'T EXIST YET : profile schema](http://google.com). Based on the defined profile schema,
```

Comment:
Stubs like this should not be used. All the pages should be available as stubs themselves now, so you should be able adding link to them directly. 
Even if that is not the case, it's better to have a broken link (such as URI with mistake) than a link to Alphabet.
Stubs in documentation tend to stay unchanged forever.